framework/auth/views.py

```diff
@@ -51,88 +57,134 @@ def reset_password(auth, **kwargs):
         }
         raise HTTPError(400, data=error_data)
 
-    if request.method == 'POST' and form.validate():
-        # new random verification key, allows CAS to authenticate the user w/o password one time only.
-        user_obj.verification_key = security.random_string(20)
-        user_obj.set_password(form.password.data)
-        user_obj.save()
-        status.push_status_message('Password reset', kind='success', trust=False)
-        # Redirect to CAS and authenticate the user with a verification key.
-        return redirect(cas.get_login_url(
-            web_url_for('user_account', _absolute=True),
-            auto=True,
-            username=user_obj.username,
-            verification_key=user_obj.verification_key
-        ))
-
-    forms.push_errors_to_status(form.errors)
+    if request.method == 'GET':
```

Comment:
Note for CR: explicit `if` flow for better understanding. Same for `forgot_password`.